intezer: add polling commands

matany90 opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
- Add polling option to file, url and endpoint analysis commands
- Replace generic polling with polling command in Playbook

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=23771) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=23771) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=23771) it.</sub>

matany90 commented:
@yaakovpraisler the build pass, can you please review?

yaakovpraisler commented on Packs/Intezer/Integrations/IntezerV2/IntezerV2.py:
```diff
@@ -656,6 +801,46 @@ def _refine_gene_counts(summary: dict):
+    zip_password = args.get('zip_password')
+    disable_dynamic_execution = args.get('disable_dynamic_execution')
+    disable_static_extraction = args.get('disable_static_extraction')
+    sandbox_command_line_arguments = args.get('sandbox_command_line_arguments')
```
I can't find the `sandbox_command_line_arguments` argument in the yaml file

yaakovpraisler commented on Packs/Intezer/Playbooks/playbook-Intezer_-_Analyze_File_and_URL.yml:
```diff
@@ -1,6 +1,6 @@
 id: Intezer - Analyze File and URL
 version: -1
-fromversion: 6.5.0
+fromversion: 6.2.0
```
@efelmandar - Do we support this?

matany90 replied:
Probably my mistake; I changed it back to 6.5.0

yaakovpraisler commented on Packs/Intezer/ReleaseNotes/1_5_0.md:
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+##### Intezer v2
+- Support polling using schedule tasks instead of generic polling 
```
```suggestion
- Added support for *polling* to the following commands:
 - ***intezer-analyze-by-hash***
 - ***intezer-analyze-url***
 - ***intezer-analyze-by-file***
- Added the following new *polling* commands:
 - ***intezer-get-file-analysis-result***
 - ***intezer-get-url-analysis-result***
 - ***intezer-get-endpoint-analysis-result***
```

yaakovpraisler reviewed (changes requested):
Hi @matany90, Thank you for your contribution! Nice work :)

Just wanted to verify - Did you added the new result commands only to support polling in the existing commands (I ask because you can support this without adding new commands)? or you still want the result commands regardless of the existing commands?

@efelmandar - Please approve the playbook changes

yaakovpraisler commented on Packs/Intezer/Integrations/IntezerV2/IntezerV2.py:
```diff
@@ -131,7 +147,7 @@ def check_is_available(intezer_api: IntezerApi, args: dict) -> str:
 
 
-def analyze_by_hash_command(intezer_api: IntezerApi, args: Dict[str, str]) -> CommandResults:
+def analyze_by_hash_command(args: Dict[str, str], intezer_api: IntezerApi) -> PollResult:
```
Same for the `analyze_url_command` and `analyze_by_uploaded_file_command` command functions

matany90 commented:
@yaakovpraisler
Thank you for the review!

About your first question about the new commands:
I still want to add those three new commands, regardless of the polling functionality I added in "intezer-analyze-by-hash", "intezer-analyze-by-file" and "intezer-analyze-url" commands.

About the Playbook version that I changed from 6.5.0 to 6.2.0, I did it by mistake; I changed it back to 6.5.0

The rest of the issues you raised were fixed :)

yaakovpraisler reviewed (commented):
Looks good! Just a few small comments -

Please let me know when you're available so I'll schedule us a call for a demo
@kgal-pan FYI

yaakovpraisler commented:
Hi @matany90 -
We can do it on Thursday, just let me know when it's convenient for you and I'll schedule us a call.

matany90 commented:
@yaakovpraisler 
Thursday around 11:30 works for me

efelmandar commented:
Hi @matany90 and thank you for your contribution, I reviewed the changes and all looks good, great work!
